src/applications/third-party-apps/containers/SearchForm/index.jsx

```diff
@@ -0,0 +1,148 @@
+// Node modules.
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import URLSearchParams from 'url-search-params';
+import map from 'lodash/map';
+import { connect } from 'react-redux';
+// Relative imports.
+import CATEGORIES from '../../constants/CATEGORIES';
+import PLATFORMS from '../../constants/PLATFORMS';
+import scrollToTop from 'platform/utilities/ui/scrollToTop';
+import { fetchResultsThunk } from '../../actions';
+
+export class SearchForm extends Component {
+  static propTypes = {
+    // From mapStateToProps.
+    fetching: PropTypes.bool.isRequired,
+    // From mapDispatchToProps.
+    fetchResultsThunk: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    // Derive the current name params.
+    const queryParams = new URLSearchParams(window.location.search);
+
+    // Derive the state values from our query params.
+    const category = queryParams.get('category') || '';
+    const platform = queryParams.get('platform') || '';
+
+    this.state = {
+      category,
+      platform,
+    };
+  }
+
+  componentDidMount() {
+    const { category, platform } = this.state;
+
+    // Fetch the results with their name if it's on the URL.
+    if (category || platform) {
+      this.props.fetchResultsThunk({ category, platform });
+    }
+  }
+
+  onStateChange = key => event => {
+    this.setState({ [key]: event.target.value });
+  };
```

Comment:
This could be a good trial run for that whole React hook thing...Are you interesting at all in trying out React hooks in this app?

Reply:
I made it with React hooks originally, then reverted since our testing infrastructure doesn't support them 😭